### DISCUSSION TITLE
browser notification

### DISCUSSION THREAD

**Discussion by Auric-Manteo** [marimo-team/marimo#2513] [Q&A]
Hi!
Is it possible to generate a notification, something like "execution finished" as a browser notification by executing a piece of JavaScript in the browser or something similar?

I am running marimo remotely and have some long running cells at the end of which I would like to add a "notify_execution_successful()" call that shows a browser notification similar to what I get with google calendar notifications.

Since I'm running marimo remotely I can't easily use my OS desktop notifications.

**Comment by mscolnick** [comment]
We have a util `mo.status.toast` (which doesn't actually looked documented by mistake). This does not use the OS desktop notification, but does so inside the application. We would accept a PR to add an arg to promote it to a desktop notification

**Reply by joe42** [reply]
Thanks! Could you give me some advice on how to do that?
I went through the toast code with a debugger and if I understand it correctly it sends an alert to the frontend, but otherwise I'm a bit lost.
How should I add the JavaScript code to create the desktop notification to the code base?
Maybe you could link to some relevant code locations?
